src/stan/variational/advi_params_fullrank.hpp

```diff
@@ -0,0 +1,129 @@
+          stan::math::check_not_nan(function, "Mean vector", mu_(i));
+        stan::math::check_lower_triangular(function,
+                               "Cholesky factor", L_chol_);
+      }
+
+      virtual ~advi_params_fullrank() {}  // No-op
+
+      // Accessors
+      int dimension() const { return dimension_; }
+      const Eigen::VectorXd& mu()     const { return mu_; }
+      const Eigen::MatrixXd& L_chol() const { return L_chol_; }
+
+      // Mutators
+      void set_mu(const Eigen::VectorXd& mu) {
+        static const char* function =
+          "stan::variational::advi_params_fullrank::set_mu";
+
+        stan::math::check_size_match(function,
+                               "Dimension of input vector", mu.size(),
+                               "Dimension of current vector", dimension_);
+        for (int i = 0; i < dimension_; ++i)
+          stan::math::check_not_nan(function, "Input vector", mu(i));
+        mu_ = mu;
+      }
+
+      void set_L_chol(const Eigen::MatrixXd& L_chol) {
+        static const char* function =
+          "stan::variational::advi_params_fullrank::set_L_chol";
+
+        stan::math::check_size_match(function,
+                               "Dimension of mean vector",     dimension_,
+                               "Dimension of Cholesky factor", L_chol.rows());
+        stan::math::check_lower_triangular(function,
+                               "Cholesky factor", L_chol);
+
+        L_chol_ = L_chol;
+      }
+
+      // Entropy of normal:
+      // 0.5 * dim * (1+log2pi) + 0.5 * log det (L^T L) =
+      // 0.5 * dim * (1+log2pi) + sum(log(abs(diag(L))))
+      double entropy() const {
+        double tmp(0.0);
+        double result(
+          0.5 * static_cast<double>(dimension_)
+          * (1.0 + stan::prob::LOG_TWO_PI));
+        for (int d = 0; d < dimension_; ++d) {
+          tmp = fabs(L_chol_(d, d));
+          if (tmp != 0.0) {
+            result += log(tmp);
+          }
+        }
+        return result;
+      }
+
+      // // Calculate natural parameters
+      // Eigen::VectorXd nat_params() const {
+
+      //   // FIXME: stupid Eigen. can't initialize LLT factor with L.
+      //   // Compute the covariance matrix
+      //   Eigen::MatrixXd Sigma = L_chol_ * L_chol_.transpose();
+      //   stan::math::LDLT_factor<double,-1,-1> Sigma_LDLT(Sigma);
+
+      //   // Create a vector twice the dimension size
+      //   Eigen::VectorXd natural_params(dimension_ + dimension_^2);
+
+      //   // Concatenate the natural parameters
+      //   natural_params << (Sigma_LDLT.solve(mu_)).array(),
+      //                     (Sigma_LDLT.solve(
+      //                       Eigen::MatrixXd::Identity(dimension_,dimension_))
+      //                     ).array();
+
+      //   return natural_params;
+      // }
+
+      // Implements f^{-1}(\check{z}) = L\check{z} + \mu
+      Eigen::VectorXd to_unconstrained(const Eigen::VectorXd& z_check) const {
```

Comment:
@akucukelbir I think the function name 'to_unconstrained' doesn't comport with Stan's definition of unconstrained. unconstrained is about the support of the parameters, and it looks like this function just pushes MVN(0, I) to MVN(mu, Cov).